app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,544 @@
+      try {
+        LocationSettingsResponse response = task.getResult(ApiException.class);
+        // All location settings are satisfied. The client can initialize location
+        // requests here.
+
+        startService(ACTION_TURN_ON_AFTER_O);
+
+        //}
+      } catch (ApiException exception) {
+        switch (exception.getStatusCode()) {
+          case LocationSettingsStatusCodes.RESOLUTION_REQUIRED:
+            // Location settings are not satisfied. But could be fixed by showing the
+            // user a dialog.
+            try {
+              // Cast to a resolvable exception.
+              ResolvableApiException resolvable = (ResolvableApiException) exception;
+              // Show the dialog by calling startResolutionForResult(),
+              // and check the result in onActivityResult().
+              resolvable.startResolutionForResult(
+                  ZimHostActivity.this,
+                  LOCATION_SETTINGS_PERMISSION_RESULT);
+            } catch (IntentSender.SendIntentException e) {
+              // Ignore the error.
+            } catch (ClassCastException e) {
+              // Ignore, should be an impossible error.
+            }
+            break;
+          case LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE:
+            // Location settings are not satisfied. However, we have no way to fix the
+            // settings so we won't show the dialog.
+            break;
+          default:
+            break;
+        }
+      }
+    });
+  }
+
+  //Advice user to turn on hotspot manually for API<26
+  private void startHotspotDialog() {
+    AlertDialog.Builder builder = new AlertDialog.Builder(this, dialogStyle());
+
+    builder.setPositiveButton(getString(R.string.go_to_settings_label),
+        (dialog, id) -> setupWifiSettingsIntent());
+
+    builder.setNeutralButton(getString(R.string.hotspot_dialog_neutral_button), (dialog, id) -> {
+      //TO DO: START SERVER WITHIN THE SERVICE.
+      //Adding a handler because sometimes hotspot can take time to turn on.
+      //TO DO: Add a progress dialog instead of handler
+      ProgressDialog progressDialog =
+          ProgressDialog.show(this, getString(R.string.progress_dialog_starting_server), "",
+              true);
+      progressDialog.show();
+      final Handler handler = new Handler();
+      handler.postDelayed(() -> {
+        progressDialog.dismiss();
+        startService(ACTION_START_SERVER);
+      }, 7000);
+    });
+
+    builder.setTitle(getString(R.string.hotspot_dialog_title));
+    builder.setMessage(
+        getString(R.string.hotspot_dialog_message)
+    );
+    AlertDialog dialog = builder.create();
+    dialog.show();
+  }
+
+  private void startService(String ACTION) {
+    if (ACTION.equals(ACTION_START_SERVER)) {
+      serviceIntent.putStringArrayListExtra(SELECTED_ZIM_PATHS_KEY, selectedBooksPath);
+    }
+    serviceIntent.setAction(ACTION);
+    this.startService(serviceIntent);
```

Comment:
You want a foregroundService correct? Use `ContextCompat.startForegroundService`, it will start foreground on O and just call this method below O